### ISSUE TITLE
[Bug] Some videos show just error messages (NilAssertionError)

### ISSUE THREAD

**Issue by hermann-p** [iv-org/invidious#1996]
**Describe the bug**

When navigating to some videos, like [this one](https://invidious.tube/watch?v=9PJd3ltvO7Q), instead of the player page it shows an error dump (NilAssertionError) and asks me to create an issue for it. Others will play just fine.

**Logs**

    Title: `Nil assertion failed (NilAssertionError)`
    Date: `2021-04-08T11:51:15Z`
    Route: `/watch?v=9PJd3ltvO7Q`
    Version: `2021.04.02-c4c2e01 @ master`
    <details>
    <summary>Backtrace</summary>
    <p>
   
    ```
    Nil assertion failed (NilAssertionError)
      from ???
      from ???
      ...
      from ??? // total of 24x
    ```
    </p>
    </details>

**Additional information**

- Firefox 87.0
- linux 5.4.108

**Comment by jakoblm** [comment]
The same thing happens [here](https://invidious.tube/watch?v=yEoKWle-3u8): 

``````
Title: `Nil assertion failed (NilAssertionError)`
Date: `2021-04-08T15:07:19Z`
Route: `/watch?v=yEoKWle-3u8`
Version: `2021.04.02-c4c2e01 @ master`
<details>
<summary>Backtrace</summary>
<p>
   
```
Nil assertion failed (NilAssertionError)
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
  from ???
```
</p>
</details>
``````

**Comment by tenpura-shrimp** [comment]
works on other instances. Must be an instance issue.

**Comment by jakoblm** [comment]
The second one works on other instances, the first one does not. I have tried several. 
However, the message now appears there: "The media could not be loaded, either because the server or network failed or because the format is not supported."

**Comment by tenpura-shrimp** [comment]
The media could not be loaded - this is due to google ip restriciton.

You need to use proxy or dash for that.

**Comment by jakoblm** [comment]
[Here](https://invidious.xyz/watch?v=2gWVjyH_EFA) it loads even without proxy or dash. The same on some other distances. Sometimes it works, sometimes not. What could be the reason for this?

**Comment by markozajc** [comment]
I'm running version b7eba3b and all four videos listed in this thread work as of time of writing. It seems that this has been fixed since c4c2e01.
